stdlib/json.gr

```diff
@@ -0,0 +1,2157 @@
+/**
+ * JSON (JavaScript Object Notation) parsing and printing.
+ */
+
+module Json
+
+include "string"
+include "char"
+include "list"
+include "option"
+include "number"
+include "uint8"
+include "runtime/unsafe/wasmi32"
+include "runtime/unsafe/wasmi64"
+include "runtime/unsafe/wasmf32"
+include "runtime/unsafe/wasmf64"
+include "runtime/numberUtils"
+include "runtime/unsafe/tags"
+include "runtime/numbers"
+include "runtime/unsafe/memory"
+include "buffer"
+include "runtime/string" as RunTimeString
+include "runtime/numbers"
+include "runtime/unsafe/memory"
+include "runtime/dataStructures"
+include "runtime/bigint" as BI
+from RunTimeString use { toString as runtimeToString, getCodePoint }
+from Numbers use { coerceNumberToWasmI32 }
+from DataStructures use { tagSimpleNumber, newFloat64, untagSimpleNumber }
+
+// Primitive Offsets
+// TODO(703): Get these offsets from the runtime
+@unsafe
+let _INT64_BOXED_VALUE_OFFSET = 8n
+@unsafe
+let _Float64_BOXED_VALUE_OFFSET = 8n
+/**
+ * Data structure representing the result of parsing and the input of printing
+ * JSON.
+ *
+ * For example this object
+ * ```grain
+ * JSONObject([
+ *   ("currency", JSONString("€")),
+ *   ("price", JSONNumber(99.99)),
+ * ])
+ * ```
+ * is equivalent to the following JSON:
+ * ```json
+ * {"currency":"€","price":99.99}
+ * ```
+ *
+ * This data structure is semantically equivalent to the JSON format allowing
+ * mostly lossless round trips of printing and parsing. Exceptions to this are
+ * white spaces, multiple ways JSON allows escaping characters in strings and
+ * some edge-cases related to Grain's `Number` type.
```

Comment:
```suggestion
 * whitespace, multiple ways JSON allows escaping characters in strings, and
 * some edge cases related to Grain's `Number` type.
```

Reply:
This is also the documentation that will appear on the website, so we should consider how much of this is important to say here. cc @phated